fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/repository/domain/ExternalEventConfiguration.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.event.external.repository.domain;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name = "m_external_event_configuration")
+@Getter
+@NoArgsConstructor
+public class ExternalEventConfiguration {
+
+    @Id
+    @Column(name = "type", nullable = false)
+    private String type;
+
+    @Column(name = "enabled", nullable = false)
+    private boolean enabled = false;
+
+    public ExternalEventConfiguration(String type, boolean enabled) {
+        this.type = type;
+        this.enabled = enabled;
+    }
+
+    public boolean enableConfiguration(final boolean canEnable) {
```

Comment:
Not sure we wanna do this tbh. There should be nothing wrong if somebody tries to set it enabled when it's already enabled. Thoughts?